test/e2e/specs/all-features-confirmation/recordset/presentation.spec.ts

```diff
@@ -0,0 +1,804 @@
+      });
+
+      await test.step(`should have ${params.columns.length} columns`, async () => {
+        const columnTitles = params.columns.map((col) => col.title);
+        await expect.soft(RecordsetLocators.getColumnNames(page)).toHaveText(columnTitles);
+      });
+
+      await test.step('should display the Export dropdown button with proper tooltip.', async () => {
+        const exportDropdown = ExportLocators.getExportDropdown(page);
+        await expect.soft(exportDropdown).toBeVisible();
+
+        await testTooltip(exportDropdown, testParams.tooltip.exportDropdown, APP_NAMES.RECORDSET);
+      });
+
+      await testExportDropdown(page, params.file_names, APP_NAMES.RECORDSET);
+
+      await test.step('should show information icon after column name in column headers which have a comment and inspect the comment value', async () => {
+        const columnsWComments = params.columns.filter((c) => typeof c.comment === 'string');
+
+        const columnsEls = RecordsetLocators.getColumnsWithTooltipIcon(page);
+        await expect.soft(columnsEls).toHaveCount(columnsWComments.length);
+
+        const testColumnTooltip = async (idx: number) => {
+
+          // if we reached the end of the list, then finish the test case
+          if (idx === columnsWComments.length) return;
+
+          const comment = columnsWComments[idx].comment;
+
+          await testTooltip(columnsEls.nth(idx), comment, APP_NAMES.RECORDSET, true)
+          await testColumnTooltip(idx + 1);
+        }
+
+        // go one by one over the columns w comments and test their tooltip
+        await testColumnTooltip(0);
+      });
+
+      await test.step('have correct tooltip for action column', async () => {
+        await testTooltip(RecordsetLocators.getActionsHeader(page), testParams.tooltip.actionCol, APP_NAMES.RECORDSET, true);
+      });
+
+      await test.step('apply different searches', async () => {
+        // apply simple search words
+        await testMainSearch(page, 'Super 8 North Hollywood Motel', 1);
+
+        // apply conjunctive search words
+        await testMainSearch(page, '"Super 8" motel "North Hollywood"', 1);
+
+        // apply non-matching search words
+        await testMainSearch(page, 'asdfghjkl', 0);
+      });
+
+      await test.step('JSON Column value should be searchable', async () => {
+        const searchBox = RecordsetLocators.getMainSearchInput(page),
+          searchSubmitButton = RecordsetLocators.getSearchSubmitButton(page);
+
+        // search for a row that is not the first one after sorting
+        await searchBox.fill('9876.3543');
+        await searchSubmitButton.click();
+        await RecordsetLocators.waitForRecordsetPageReady(page);
+
+        await expect.soft(RecordsetLocators.getRows(page)).toHaveCount(1);
+        // clear search in next it case
+      });
+
+      await test.step('check the link of the view details after searching', async () => {
```

Comment:
I feel like it might be better if this step and the previous one are merged into one. Although this is completely subjective.